# Denad code along

Note: Trying to import statsmodels and having it work is weird. Here is the process:

Create a new environment:<br>
conda create -n env-name-here

Install statsmodels:<br>
conda install -c conda-forge statsmodels

Now go to Anaconda Navigator -> Environments -> env-name-here -> Start Button -> Open Terminal -> Enter jupyter notebook -> Navigate to your jupyter notebook file.

For some reason, if you use anaconda prompt, use `conda activate env-name-here`, enter `jupyter notebook`, then try to import statsmodels, *it won't work*. I've settled for the easier approach until I can make this latter approach work.

Found second command on https://www.statsmodels.org/dev/install.html

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import statsmodels.formula.api as smf

pd.set_option('display.max_columns', None)

In [2]:
# Import data
df_T = pd.read_csv('./data/twin_pairs_T_3years_samesex.csv')
df_X = pd.read_csv('./data/twin_pairs_X_3years_samesex.csv')
df_Y = pd.read_csv('./data/twin_pairs_Y_3years_samesex.csv')

In [3]:
df_T.head()

Unnamed: 0  dbirwt_0  dbirwt_1
0           0    2268.0    2296.0
1           1    2610.0    2650.0
2           2    1985.0    2098.0
3           3    2410.0    2420.0
4           4    2013.0    2637.0

In [4]:
df_X.head()

Unnamed: 0.1  Unnamed: 0  pldel  birattnd  brstate  stoccfipb  mager8  \
0             0           0    1.0       1.0        1          1       3   
1             1           1    1.0       1.0        1          1       3   
2             2           2    1.0       1.0        1          1       5   
3             3           3    1.0       1.0        1          1       4   
4             4           4    1.0       1.0        1          1       5   

   ormoth  mrace  meduc6  dmar  mplbir  mpre5  adequacy  orfath  frace  \
0     0.0      1     NaN     1    31.0    1.0       1.0     0.0    1.0   
1     0.0      2     3.0     0     1.0    1.0       1.0     NaN    NaN   
2     0.0      1     5.0     1    59.0    1.0       1.0     0.0    1.0   
3     0.0      2     1.0     1     1.0    2.0       2.0     0.0    2.0   
4     0.0      2     4.0     0     1.0    1.0       1.0     NaN    NaN   

   birmon  gestat10  csex  anemia  cardiac  lung  diabetes  herpes  hydra  \
0       1         9     1     0.0      0.0   0.0       0.0     0.0    0.0   
1       1         6     1     0.0      0.0   0.0       0.0     0.0    0.0   
2       1         4     0     0.0      0.0   0.0       0.0     0.0    0.0   
3       1         6     0     0.0      0.0   0.0       0.0     0.0    0.0   
4       1         6     0     0.0      0.0   0.0       0.0     0.0    0.0   

   hemo  chyper  phyper  eclamp  incervix  pre4000  preterm  renal   rh  \
0   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
1   0.0     0.0     1.0     0.0       0.0      0.0      0.0    0.0  0.0   
2   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
3   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
4   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   

   uterine  othermr  tobacco  alcohol  cigar6  drink5  crace  data_year  \
0      0.0      0.0      0.0      0.0     0.0     0.0      1         89   
1      0.0      0.0      0.0      0.0     0.0     0.0      2         89   
2      0.0      0.0      0.0      0.0     0.0     0.0      1         89   
3      0.0      0.0      0.0      0.0     0.0     0.0      2         89   
4      0.0      0.0      0.0      0.0     0.0     0.0      2         89   

   nprevistq  dfageq  feduc6  infant_id_0  infant_id_1  dlivord_min  \
0        4.0     0.0     NaN            1            0          NaN   
1        3.0     NaN     NaN            7            6          2.0   
2        1.0     6.0     3.0           11           10          NaN   
3        1.0     6.0     1.0           15           14          4.0   
4        2.0     NaN     NaN           23           22          NaN   

   dtotord_min  bord_0  bord_1  brstate_reg  stoccfipb_reg  mplbir_reg  
0          NaN     NaN     NaN            5              5         1.0  
1          3.0     2.0     1.0            5              5         5.0  
2          NaN     NaN     NaN            5              5         1.0  
3          4.0     1.0     2.0            5              5         5.0  
4          NaN     NaN     NaN            5              5         5.0

In [5]:
df_Y.head()

Unnamed: 0  mort_0  mort_1
0           0     0.0     0.0
1           1     0.0     0.0
2           2     0.0     0.0
3           3     0.0     0.0
4           4     0.0     0.0

From the ReadmeTwins file we notice the following about the data:
- X - includes 50 covariates for the twin pair. _0 and _1 denote differences in sex and birth order. The covariates pertain to parent characteristics, both mother and father.
- Y - mortality outcomes for twins, given by `mort_0` and `mort_1` respectively
- T - birth weights in grams for both twins

In [6]:
for df in [df_X,df_Y,df_T]:
    print(str(df.info()) + "\n\n\n" + "=====================================================")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71345 entries, 0 to 71344
Data columns (total 54 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0.1   71345 non-null  int64  
 1   Unnamed: 0     71345 non-null  int64  
 2   pldel          71321 non-null  float64
 3   birattnd       71231 non-null  float64
 4   brstate        71345 non-null  int64  
 5   stoccfipb      71345 non-null  int64  
 6   mager8         71345 non-null  int64  
 7   ormoth         69319 non-null  float64
 8   mrace          71345 non-null  int64  
 9   meduc6         67006 non-null  float64
 10  dmar           71345 non-null  int64  
 11  mplbir         71237 non-null  float64
 12  mpre5          69718 non-null  float64
 13  adequacy       68176 non-null  float64
 14  orfath         59523 non-null  float64
 15  frace          61085 non-null  float64
 16  birmon         71345 non-null  int64  
 17  gestat10       71345 non-null  int64  
 18  csex  

An interesting problem: can we predict what causally influences mortality within the twins?

To do this, we can apply linear regression here.

First, create a df that contains mortality and regressors.

In [7]:
df_X.shape

(71345, 54)

In [8]:
df_Y.shape

(71345, 3)

In [9]:
mort_df = pd.concat([df_X,df_T,df_Y],axis=1)
mort_df

Unnamed: 0.1  Unnamed: 0  pldel  birattnd  brstate  stoccfipb  mager8  \
0                 0           0    1.0       1.0        1          1       3   
1                 1           1    1.0       1.0        1          1       3   
2                 2           2    1.0       1.0        1          1       5   
3                 3           3    1.0       1.0        1          1       4   
4                 4           4    1.0       1.0        1          1       5   
...             ...         ...    ...       ...      ...        ...     ...   
71340         71340       71340    1.0       1.0       51         56       4   
71341         71341       71341    1.0       1.0       51         56       4   
71342         71342       71342    1.0       1.0       51         56       2   
71343         71343       71343    1.0       1.0       51         56       5   
71344         71344       71344    1.0       1.0       51         56       5   

       ormoth  mrace  meduc6  dmar  mplbir  mpre5  adequacy  orfath  frace  \
0         0.0      1     NaN     1    31.0    1.0       1.0     0.0    1.0   
1         0.0      2     3.0     0     1.0    1.0       1.0     NaN    NaN   
2         0.0      1     5.0     1    59.0    1.0       1.0     0.0    1.0   
3         0.0      2     1.0     1     1.0    2.0       2.0     0.0    2.0   
4         0.0      2     4.0     0     1.0    1.0       1.0     NaN    NaN   
...       ...    ...     ...   ...     ...    ...       ...     ...    ...   
71340     0.0      3     4.0     0    51.0    1.0       1.0     0.0    3.0   
71341     0.0      3     3.0     0    51.0    3.0       3.0     0.0    3.0   
71342     0.0      1     2.0     1    38.0    1.0       2.0     0.0    1.0   
71343     0.0      1     5.0     1     6.0    1.0       1.0     0.0    1.0   
71344     0.0      1     3.0     1    51.0    1.0       1.0     0.0    1.0   

       birmon  gestat10  csex  anemia  cardiac  lung  diabetes  herpes  hydra  \
0           1         9     1     0.0      0.0   0.0       0.0     0.0    0.0   
1           1         6     1     0.0      0.0   0.0       0.0     0.0    0.0   
2           1         4     0     0.0      0.0   0.0       0.0     0.0    0.0   
3           1         6     0     0.0      0.0   0.0       0.0     0.0    0.0   
4           1         6     0     0.0      0.0   0.0       0.0     0.0    0.0   
...       ...       ...   ...     ...      ...   ...       ...     ...    ...   
71340      11         6     1     0.0      0.0   0.0       0.0     0.0    0.0   
71341      11         6     1     0.0      0.0   0.0       0.0     0.0    0.0   
71342      11         4     1     0.0      0.0   0.0       0.0     0.0    0.0   
71343      12         4     1     0.0      0.0   0.0       0.0     0.0    0.0   
71344      12         6     0     0.0      0.0   0.0       0.0     0.0    0.0   

       hemo  chyper  phyper  eclamp  incervix  pre4000  preterm  renal   rh  \
0       0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
1       0.0     0.0     1.0     0.0       0.0      0.0      0.0    0.0  0.0   
2       0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
3       0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
4       0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
...     ...     ...     ...     ...       ...      ...      ...    ...  ...   
71340   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
71341   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
71342   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
71343   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   
71344   0.0     0.0     0.0     0.0       0.0      0.0      0.0    0.0  0.0   

       uterine  othermr  tobacco  alcohol  cigar6  drink5  crace  data_year  \
0          0.0      0.0      0.0      0.0     0.0     0.0      1         89   
1          0.0      0.0      

In [10]:
mort_df.drop(["Unnamed: 0", "Unnamed: 0.1"],axis=1,inplace=True)
mort_df

pldel  birattnd  brstate  stoccfipb  mager8  ormoth  mrace  meduc6  \
0        1.0       1.0        1          1       3     0.0      1     NaN   
1        1.0       1.0        1          1       3     0.0      2     3.0   
2        1.0       1.0        1          1       5     0.0      1     5.0   
3        1.0       1.0        1          1       4     0.0      2     1.0   
4        1.0       1.0        1          1       5     0.0      2     4.0   
...      ...       ...      ...        ...     ...     ...    ...     ...   
71340    1.0       1.0       51         56       4     0.0      3     4.0   
71341    1.0       1.0       51         56       4     0.0      3     3.0   
71342    1.0       1.0       51         56       2     0.0      1     2.0   
71343    1.0       1.0       51         56       5     0.0      1     5.0   
71344    1.0       1.0       51         56       5     0.0      1     3.0   

       dmar  mplbir  mpre5  adequacy  orfath  frace  birmon  gestat10  csex  \
0         1    31.0    1.0       1.0     0.0    1.0       1         9     1   
1         0     1.0    1.0       1.0     NaN    NaN       1         6     1   
2         1    59.0    1.0       1.0     0.0    1.0       1         4     0   
3         1     1.0    2.0       2.0     0.0    2.0       1         6     0   
4         0     1.0    1.0       1.0     NaN    NaN       1         6     0   
...     ...     ...    ...       ...     ...    ...     ...       ...   ...   
71340     0    51.0    1.0       1.0     0.0    3.0      11         6     1   
71341     0    51.0    3.0       3.0     0.0    3.0      11         6     1   
71342     1    38.0    1.0       2.0     0.0    1.0      11         4     1   
71343     1     6.0    1.0       1.0     0.0    1.0      12         4     1   
71344     1    51.0    1.0       1.0     0.0    1.0      12         6     0   

       anemia  cardiac  lung  diabetes  herpes  hydra  hemo  chyper  phyper  \
0         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
1         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     1.0   
2         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
3         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
4         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
...       ...      ...   ...       ...     ...    ...   ...     ...     ...   
71340     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71341     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71342     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71343     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71344     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   

       eclamp  incervix  pre4000  preterm  renal   rh  uterine  othermr  \
0         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
1         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
2         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
3         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
4         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
...       ...       ...      ...      ...    ...  ...      ...      ...   
71340     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71341     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71342     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71343     0.0       0.0      0.0      0.0    0.0  0.0      1.0      1.0   
71344     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   

       tobacco  alcohol  cigar6  drink5  crace  data_year  nprevistq  dfageq  \
0          0.0      0.0     0.0     0.0      1         89        4.0     0.0   
1          0.0      0.0     0.0     0.0      2         89        3.0     NaN   
2          0.0      0.0     0.0     0.0    

Can we get the number of mortalities for the lower weighted vs. higher weighted twins?

In [11]:
mort_df.shape[0]

71345

In [12]:
# Both twins survive
mort_df.loc[(mort_df['mort_0']==0) & (mort_df['mort_1']==0),["mort_0","mort_1"]].shape[0]

67755

In [13]:
# The second twin survives
mort_df.loc[(mort_df['mort_0']==1) & (mort_df['mort_1']==0),["mort_0","mort_1"]].shape[0]

1282

In [14]:
# The first twin survives
mort_df.loc[(mort_df['mort_0']==0) & (mort_df['mort_1']==1),["mort_0","mort_1"]].shape[0]

774

In [15]:
# Neither survives
mort_df.loc[(mort_df['mort_0']==1) & (mort_df['mort_1']==1),["mort_0","mort_1"]].shape[0]

1534

What would we expect to use as the 'treatment' in this case? What about birth weights? This is given by `dbirwt_0` and `dbirwt_1`.

What we might do is to create a new column with a difference in weight, and use that as the treatment.

In [16]:
mort_df['dbirwt_diff'] = mort_df['dbirwt_1'] - mort_df['dbirwt_0']
mort_df.drop(['dbirwt_1','dbirwt_0'],axis=1,inplace=True)

What can we do given this information?

Combine these into averages. If we take the average number of twins that survive for each observation, we can see how likely it is for both to survive

In [17]:
# This is what I tried previously
# mort_col = (mort_df["mort_0"].astype("bool") | mort_df["mort_1"].astype("bool")).astype("float")

In [18]:
mort_col = (mort_df["mort_0"] + mort_df["mort_1"])/2

In [19]:
mort_col.value_counts()

0.0    67755
0.5     2056
1.0     1534
Name: count, dtype: int64

In [20]:
mort_df = mort_df.drop(["mort_0","mort_1"],axis=1)

In [21]:
# Combine into df
mort_df['mort_col'] = mort_col

In [22]:
mort_df

pldel  birattnd  brstate  stoccfipb  mager8  ormoth  mrace  meduc6  \
0        1.0       1.0        1          1       3     0.0      1     NaN   
1        1.0       1.0        1          1       3     0.0      2     3.0   
2        1.0       1.0        1          1       5     0.0      1     5.0   
3        1.0       1.0        1          1       4     0.0      2     1.0   
4        1.0       1.0        1          1       5     0.0      2     4.0   
...      ...       ...      ...        ...     ...     ...    ...     ...   
71340    1.0       1.0       51         56       4     0.0      3     4.0   
71341    1.0       1.0       51         56       4     0.0      3     3.0   
71342    1.0       1.0       51         56       2     0.0      1     2.0   
71343    1.0       1.0       51         56       5     0.0      1     5.0   
71344    1.0       1.0       51         56       5     0.0      1     3.0   

       dmar  mplbir  mpre5  adequacy  orfath  frace  birmon  gestat10  csex  \
0         1    31.0    1.0       1.0     0.0    1.0       1         9     1   
1         0     1.0    1.0       1.0     NaN    NaN       1         6     1   
2         1    59.0    1.0       1.0     0.0    1.0       1         4     0   
3         1     1.0    2.0       2.0     0.0    2.0       1         6     0   
4         0     1.0    1.0       1.0     NaN    NaN       1         6     0   
...     ...     ...    ...       ...     ...    ...     ...       ...   ...   
71340     0    51.0    1.0       1.0     0.0    3.0      11         6     1   
71341     0    51.0    3.0       3.0     0.0    3.0      11         6     1   
71342     1    38.0    1.0       2.0     0.0    1.0      11         4     1   
71343     1     6.0    1.0       1.0     0.0    1.0      12         4     1   
71344     1    51.0    1.0       1.0     0.0    1.0      12         6     0   

       anemia  cardiac  lung  diabetes  herpes  hydra  hemo  chyper  phyper  \
0         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
1         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     1.0   
2         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
3         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
4         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
...       ...      ...   ...       ...     ...    ...   ...     ...     ...   
71340     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71341     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71342     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71343     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71344     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   

       eclamp  incervix  pre4000  preterm  renal   rh  uterine  othermr  \
0         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
1         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
2         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
3         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
4         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
...       ...       ...      ...      ...    ...  ...      ...      ...   
71340     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71341     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71342     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71343     0.0       0.0      0.0      0.0    0.0  0.0      1.0      1.0   
71344     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   

       tobacco  alcohol  cigar6  drink5  crace  data_year  nprevistq  dfageq  \
0          0.0      0.0     0.0     0.0      1         89        4.0     0.0   
1          0.0      0.0     0.0     0.0      2         89        3.0     NaN   
2          0.0      0.0     0.0     0.0    

Try this model:
    $$mort_i=\beta_0 + \beta_1 * diff_i + u_i$$

In [23]:
model_1 = smf.ols('mort_col ~ dbirwt_diff',data=mort_df).fit()
model_1.summary().tables[1]

<class 'statsmodels.iolib.table.SimpleTable'>

In [24]:
mort_df['mort_col'].describe()

count    71345.000000
mean         0.035910
std          0.165579
min          0.000000
25%          0.000000
50%          0.000000
75%          0.000000
max          1.000000
Name: mort_col, dtype: float64

In [25]:
mort_df['dbirwt_diff'].describe()

count    71345.000000
mean       283.849688
std        251.996141
min          1.000000
25%        100.000000
50%        220.000000
75%        397.000000
max       6350.000000
Name: dbirwt_diff, dtype: float64

What can be said about this?
- Per gram increase in weight, the probability of surviving increases by 0.000051, or .0051%
    - Thus, for 284 grams (the mean) for instance, the probability of surviving increases by 1.45% 

Lets try another model - this time, lets include all the variables.

In [26]:
mort_df

pldel  birattnd  brstate  stoccfipb  mager8  ormoth  mrace  meduc6  \
0        1.0       1.0        1          1       3     0.0      1     NaN   
1        1.0       1.0        1          1       3     0.0      2     3.0   
2        1.0       1.0        1          1       5     0.0      1     5.0   
3        1.0       1.0        1          1       4     0.0      2     1.0   
4        1.0       1.0        1          1       5     0.0      2     4.0   
...      ...       ...      ...        ...     ...     ...    ...     ...   
71340    1.0       1.0       51         56       4     0.0      3     4.0   
71341    1.0       1.0       51         56       4     0.0      3     3.0   
71342    1.0       1.0       51         56       2     0.0      1     2.0   
71343    1.0       1.0       51         56       5     0.0      1     5.0   
71344    1.0       1.0       51         56       5     0.0      1     3.0   

       dmar  mplbir  mpre5  adequacy  orfath  frace  birmon  gestat10  csex  \
0         1    31.0    1.0       1.0     0.0    1.0       1         9     1   
1         0     1.0    1.0       1.0     NaN    NaN       1         6     1   
2         1    59.0    1.0       1.0     0.0    1.0       1         4     0   
3         1     1.0    2.0       2.0     0.0    2.0       1         6     0   
4         0     1.0    1.0       1.0     NaN    NaN       1         6     0   
...     ...     ...    ...       ...     ...    ...     ...       ...   ...   
71340     0    51.0    1.0       1.0     0.0    3.0      11         6     1   
71341     0    51.0    3.0       3.0     0.0    3.0      11         6     1   
71342     1    38.0    1.0       2.0     0.0    1.0      11         4     1   
71343     1     6.0    1.0       1.0     0.0    1.0      12         4     1   
71344     1    51.0    1.0       1.0     0.0    1.0      12         6     0   

       anemia  cardiac  lung  diabetes  herpes  hydra  hemo  chyper  phyper  \
0         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
1         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     1.0   
2         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
3         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
4         0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
...       ...      ...   ...       ...     ...    ...   ...     ...     ...   
71340     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71341     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71342     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71343     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   
71344     0.0      0.0   0.0       0.0     0.0    0.0   0.0     0.0     0.0   

       eclamp  incervix  pre4000  preterm  renal   rh  uterine  othermr  \
0         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
1         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
2         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
3         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
4         0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
...       ...       ...      ...      ...    ...  ...      ...      ...   
71340     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71341     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71342     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   
71343     0.0       0.0      0.0      0.0    0.0  0.0      1.0      1.0   
71344     0.0       0.0      0.0      0.0    0.0  0.0      0.0      0.0   

       tobacco  alcohol  cigar6  drink5  crace  data_year  nprevistq  dfageq  \
0          0.0      0.0     0.0     0.0      1         89        4.0     0.0   
1          0.0      0.0     0.0     0.0      2         89        3.0     NaN   
2          0.0      0.0     0.0     0.0    

In [27]:
controls = mort_df.drop(['dbirwt_diff','mort_col'],axis=1)

In [28]:
model_2 = smf.ols('mort_col ~  dbirwt_diff+' + '+'.join(controls), data=mort_df).fit()
model_2.summary().tables[1]

<class 'statsmodels.iolib.table.SimpleTable'>

In [30]:
df_X.data_year.value_counts()

data_year
91    24557
90    24124
89    22664
Name: count, dtype: int64